src/EnergyPlus/SetPointManager.hh

```diff
@@ -955,240 +505,77 @@ namespace SetPointManager {
-        SchSetPtMgr.deallocate();                 // Array for Scheduled Setpoint Manager data
-        DualSchSetPtMgr.deallocate();             // Dual Scheduled Setpoint Manager data
-        OutAirSetPtMgr.deallocate();              // Array for Outside Air Setpoint Manager data
-        SingZoneRhSetPtMgr.deallocate();          // Array for SZRH Set Pt Mgr
-        SingZoneHtSetPtMgr.deallocate();          // Array for SZ Heating Set Pt Mgr
-        SingZoneClSetPtMgr.deallocate();          // Array for SZ Cooling Set Pt Mgr
-        SZMinHumSetPtMgr.deallocate();            // Array for SZ Min Hum Set Pt Mgr
-        SZMaxHumSetPtMgr.deallocate();            // Array for SZ Max Hum Set Pt Mgr
-        MixedAirSetPtMgr.deallocate();            // Array for Mixed Air Set Pt Mgr
-        OAPretreatSetPtMgr.deallocate();          // Array for OA Pretreat Set Pt Mgr
-        WarmestSetPtMgr.deallocate();             // Array for Warmest Set Pt Mgr
-        ColdestSetPtMgr.deallocate();             // Array for Coldest Set Pt Mgr
-        WarmestSetPtMgrTempFlow.deallocate();     // Array for Warmest Set Pt Mgr
-        RABFlowSetPtMgr.deallocate();             // Array for return air bypass
-        MZAverageCoolingSetPtMgr.deallocate();    // Array for MultiZone
-        MZAverageHeatingSetPtMgr.deallocate();    // Array for MultiZone
-        MZAverageMinHumSetPtMgr.deallocate();     // Array for MultiZone
-        MZAverageMaxHumSetPtMgr.deallocate();     // Array for MultiZone
-        MZMinHumSetPtMgr.deallocate();            // Multizone min humidity rat Set Pt Mgr
-        MZMaxHumSetPtMgr.deallocate();            // Multizone max humidity rat Set Pt Mgr
-        FollowOATempSetPtMgr.deallocate();        // Array for Follow Outdoor Air
-        FollowSysNodeTempSetPtMgr.deallocate();   // Array for Follow System
-        GroundTempSetPtMgr.deallocate();          // Array for Ground Temp Setpoint
-        CondEntSetPtMgr.deallocate();             // Condenser Entering Water Set Pt Mgr
-        IdealCondEntSetPtMgr.deallocate();        // Ideal Condenser Entering Set Pt Mgr
-        SZOneStageCoolingSetPtMgr.deallocate();   // single zone 1 stage cool
-        SZOneStageHeatingSetPtMgr.deallocate();   // single zone 1 stage heat
-        ReturnWaterResetChWSetPtMgr.deallocate(); // return water reset
-        ReturnWaterResetHWSetPtMgr.deallocate();  // hot-water return water reset
-        SchTESSetPtMgr.deallocate();              // TES Scheduled setpoint Managers
-        SystemNodeResetSetPtMgr.deallocate();     // Array for System Node Reset Temperature and Humidity Ratio setpoint manager
-
-        NoSurfaceGroundTempObjWarning = true;
-        NoShallowGroundTempObjWarning = true;
-        NoDeepGroundTempObjWarning = true;
-        NoFCGroundTempObjWarning = true;
+
+        for (int iSPM = 1; iSPM <= (int)spms.size(); ++iSPM) {
+            delete spms[iSPM];
+        }
+        spms.deallocate();
```

Comment:
`std::unique_ptr` would be unnecesary if using loop. Why use the more complicated feature when the simpler one will do?

Reply:
I've read your response above and I get it (https://github.com/NREL/EnergyPlus/pull/10537#discussion_r1624433855)

---

It's getting philosophical perhaps, but my opinion is that unique_ptr (and other smart pointers) have been created for a reason. They are there to avoid the numerous leaks that were common currency in older codebases. Your argumentation is one that's shared by a lot of people though: why would I need this new feature when my code is already 100% correct and I'm managing memory perfectly? Which typically goes with "why would I pay for overhead when I know what I'm doing" (and overhead there can be indeed!). I can understand both, perhaps I'm just not confident enough to assume I'll never do something stupid and leak. I also don't feel that this is more complicated in this case, and the fact that I no longer have to worry about memory removes a cognitive burden so it's a net win for me _personally_.

Reply:
I don't want to sound like the C-blowhard that I have been accused of being. I am fine with using C++ (and standard library) features when they are needed. There are many C++ features I wish we would use more, starting with ... you know ... polymorphism! I am less fine with using C++ features (or any features) just because it feels good to do it. EnergyPlus is not a venue for people to experiment with cool looking C++ features (if you ever suspect me of doing such a thing, please call me on it!).

You find the additional layer of `std::unique_ptr` to be comforting. I find it to be annoying especially when implementing polymorphism and needing to use `dynamic_cast`. To me, having a `delete` loop in a destructor is simpler and more comforting than toting the `unique_ptr` wrapper around, no matter how thin it is. 

Maybe I spent too many years programming in C and just got acclimated to doing explicit memory management so I don't find it to be a cognitive burden.  I will say that I believe that the EnergyPlus team in general has too little familiarity with memory management and memory layout and that this manifests as numerous tiny inefficiencies throughout the code (misuse of references creating either unnecessary copies or unnecessary indirection, poor data structure design) whose cumulative effect is probably no more than 10-15% and whose chief impact is to upset me every time I come across one. TL;DR I find having to think about memory management (to the point where you get good at it and it becomes a mindless habit) to be a feature. I would say that I am surprised that given how much development you have done you don't feel the same way. I suspect that some of this has to do with the fact that you've done a lot of development in code that does a lot of dynamic memory management (e.g., OpenStudio) where the cognitive load of memory management is far greater and features like `unique_ptr` and `shared_ptr` are much more useful. Incidentally, this is one reason why I would balk at combining EnergyPlus and OpenStudio into a single code base. I just think that EnergyPlus' current memory management model of allocate/compute/tear-down is too valuable of a simplification to abandon for uncertain gains.